### PR TITLE
Issue #141 - Move search providers into component

### LINKED ISSUE #141: Search Providers | Refactor HTML file and JS code into a component
Similarly to how was done for the on #117, encapsulate all code related to Search Providers tab into a Svelte component (i.e. move all HTML code from `views/providers.html` and all related JS code from `options.js` to a new `components/options/providers/main.svelte` file).

Blocked by #138, #139 and #140

### PR DESCRIPTION
**Pull request by Saidmurod1996**
Fixes #141 pix-649

To be merged after PR #148

### CODE REVIEW

**Comment by tim-kodirov** [comment]
LGTM

File: src/components/options/providers/main.svelte [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    bind:this="{contextMenuComponent}"
+    on:updateMainConfiguration="{mainConfigurationUpdated}" />
+
+  <h2 class="mt-3">Manage Groups</h2>
+  <Groups
+    bind:this="{groupComponent}"
+    {...$$props}
+    on:updateMainConfiguration="{mainConfigurationUpdated}"
+    on:updateMainConfiguration="{updateForms}"
+    on:updateMainConfiguration="{updateProvidersForm}" />
```
Comment (jfmdev) [inline src/components/options/providers/main.svelte:43]: This code is very confusing to me: you are adding _three_ listeners to the `updateMainConfiguration` event to execute `mainConfigurationUpdated`, `updateForms` and `updateProvidersForm` **but** `mainConfigurationUpdated` already calls `updateForms` (within is code) and `updateForms` already calls `updateProvidersForms` (within his code).

Are you sure that the only listener here shouldn't be `on:updateMainConfiguration="{mainConfigurationUpdated}"`?

File: src/components/options/settings/main.svelte [hunk 2]
```diff
@@ -29,7 +27,7 @@ function mainConfigurationUpdated(lazy) {
   if (!lazy) {
     updateLastConfigUpdate();
   }
-  dispatch("updateMainConfiguration", lazy);
+  chrome.runtime.sendMessage({ action: "updateContextualMenu" });
```
Comment (jfmdev) [inline src/components/options/settings/main.svelte:30]: With this change, you are doing the equivalent to _always_ doing `dispatch("updateMainConfiguration", true);`, **however** when the `updateNow` function is invoced, `lazy` will be False, hence I have the _feeling_ (I haven't tried yet) that the UI component for providers isn't being updated when the users clicks the "Update now" button.

**Review by jfmdev — CHANGES_REQUESTED** [review]

File: src/components/options/providers/main.svelte (hunk 1 above)
Comment (Saidmurod1996) [reply src/components/options/providers/main.svelte:43]: Yes you are right, after changing the function I forgot to remove the listeners.

Comment (jfmdev) [reply src/components/options/providers/main.svelte:43]: Did you committed your changes? The three listeners seems to still be there...

File: src/components/options/providers/main.svelte [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  <h2 class="mt-4">Manage Context Menu Items</h2>
+  <ContextMenuItems
+    bind:this="{contextMenuComponent}"
+    on:updateMainConfiguration="{mainConfigurationUpdated}" />
+
+  <h2 class="mt-3">Manage Groups</h2>
+  <Groups
+    bind:this="{groupComponent}"
+    {...$$props}
```
[the remark this answers is not in the document]
Comment (jfmdev) [reply src/components/options/providers/main.svelte:40]: This was marked as resolved, but I don't see any change on the code nor justification.

**Comment by Saidmurod1996** [comment]
I remember solving those comments, but somehow they are unstaged from commit🤦🏾‍♂️

**Review by jfmdev — APPROVED** [review]

**Pull request merged**